stacker/dag/__init__.py

```diff
@@ -0,0 +1,313 @@
+import threading
+import logging
+from copy import copy, deepcopy
+from collections import deque
+
+logger = logging.getLogger(__name__)
+
+try:
+    from collections import OrderedDict
+except:
+    from ordereddict import OrderedDict
+
+
+class DAGValidationError(Exception):
+    pass
+
+
+class DAG(object):
```

Comment:
I think `transpose` is generic enough to contribute back. `walk` is somewhat specific, being threaded and all (in reality, I should probably pull it off of `DAG` entirely and just make it a function that takes a `DAG` as an argument).